components/filter/demo/filter.html

```diff
@@ -0,0 +1,79 @@
+						<d2l-filter-dimension-set-value key="biology" text="Biology"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="chemistry" text="Chemistry"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="drama" text="Drama"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="english" text="English"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="how-to" text="How To Write a How To Article With a Flashy Title"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="math" text="Math"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="physics" text="Physics"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="stats" text="Statistics"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="writerscraft" text="Writer's Craft"></d2l-filter-dimension-set-value>
+					</d2l-filter-dimension-set>
+				</d2l-filter>
+			</template>
+		</d2l-demo-snippet>
+
+		<h2>Multiple Set Dimensions</h2>
+		<d2l-demo-snippet>
+			<template>
+				<d2l-filter>
+					<d2l-filter-dimension-set key="course" text="Course">
+						<d2l-filter-dimension-set-value key="art" text="Art"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="astronomy" text="Astronomy"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="biology" text="Biology"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="chemistry" text="Chemistry"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="drama" text="Drama"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="english" text="English"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="how-to" text="How To Write a How To Article With a Flashy Title"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="math" text="Math"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="physics" text="Physics"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="stats" text="Statistics"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="writerscraft" text="Writer's Craft"></d2l-filter-dimension-set-value>
+					</d2l-filter-dimension-set>
+					<d2l-filter-dimension-set key="role" text="Role">
+						<d2l-filter-dimension-set-value key="admin" text="Admin"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="instructor" text="Instructor"></d2l-filter-dimension-set-value>
+						<d2l-filter-dimension-set-value key="student" text="Student"></d2l-filter-dimension-set-value>
+					</d2l-filter-dimension-set>
+				</d2l-filter>
+			</template>
+		</d2l-demo-snippet>
+	</d2l-demo-page>
+
+	<script type="module">
```

Comment:
Nit: I think this could become:

```javascript
<script>
	document.addEventListener('d2l-filter-change', e => {
		console.log(`Filter selection changed for dimension "${e.detail.dimension}":`, e.detail.value);
	});
</script>
```

The `<script>` could also be placed in the demo-snippet's `template` for purpose of example, but in that case care must be taken to wire up to the correct `d2l-filter`.

Reply:
So I actually had it this way because I currently have the event set not to bubble: https://github.com/BrightspaceUI/core/pull/1301/files#diff-371eee884788627e4694af0044b256de9257505f6ee6430c4302574842f8b441R174
I figured I'd start off more restrictive with `bubbles: false` and `composed: false` and we can always open it up later.  This may make things annoying when I get to the coordinator of filter and filter tags, depending on how we do that.  Do you think these should just be true from the start?

Reply:
Ah I see.  So `composed: false` for sure.  I tend to be less restrictive with the bubbling of events unless I have a reason to not bubble.  Most native browser events bubble and I think people usually expect them to.   As an example, `focus` does not (not sure why they made that choice... but it was decades ago), and after many years of requests, they ended up adding new `focusin`/`focusout` events that _do_ bubble.  But yeah not a huge deal if you decide not to... but changing it to bubble later would be potentially breaking. 

For the demo, even if it doesn't bubble, the demo script could capture:

```html
<script>
	document.addEventListener('d2l-filter-change', e => {
		console.log(`Filter selection changed for dimension "${e.detail.dimension}":`, e.detail.value);
	}, { capture: true });
</script>
```
But it's just demo code, so nbd.